core/sr-primitives/src/generic/digest.rs

```diff
@@ -58,33 +111,150 @@ impl<Item> traits::Digest for Digest<Item> where
-	/// Any 'non-system' digest item, opaque to the native code.
-	Other(Vec<u8>),
+	/// For all variants that include a `ConsensusEngineId`, consensus engine
+	/// implementations **MUST** ignore digests that have a `ConsensusEngineId`
+	/// that is not theirs.  Node implementations **MUST** reject digests that
+	/// have a `ConsensusEngineId` that corresponds to a consensus engine not in
+	/// use.  Node implementations **MUST** reject blocks as malformed if they
+	/// reject any of the block’s digest.  If the runtime supports this, the
+	/// node that issued the block **SHOULD** be reported as having committed
+	/// severe misbehavior and punished accordingly.  The invalid block, or its
+	/// hash, **SHOULD** constitute adequate proof of such misbehavior.
+	pub enum DigestItem<Hash, AuthorityId, SealSignature> {}
+
+	/// A digest item that can be produced by consensus engines. Consensus
+	/// engine implementations **MUST NOT** push digests not in this variant.
+	/// This **SHOULD** be detected at compile time. If it is not, the behavior
+	/// of the blockchain is undefined.
+	pub enum ConsensusDigest<SealSignature> {
+		/// Put a Seal on it. This **MUST** come after all other `DigestItem`
+		/// variants. There **MUST** be exactly one `Seal` per consensus engine,
+		/// and its `ConsensusEngineId` **MUST** be that of the consensus engine
+		/// that produced it.
+		Seal((ConsensusEngineId, SealSignature), (a, b)),
+		/// An inherent digest.
+		///
+		/// These are messages from the consensus engine to the runtime,
+		/// although the consensus engine can (and should) read them itself to
+		/// avoid code and state duplication. It is erroneous for a runtime to
+		/// produce these, but this is checked at compile time. Runtimes can
+		/// (and should) trust these, as with any other inherent. Consensus
+		/// engines MUST verify them.
+		Inherent((ConsensusEngineId, Vec<u8>), (a, b)),
+	}
+
+	/// A digest item that can be produced by runtimes. Runtime mplementations
+	/// **MUST NOT** push digests not in this variant. This **SHOULD** be
+	/// detected at compile time.  If it is not, the behavior of the blockchain
+	/// is undefined.
+	pub enum RuntimeDigest {
+		/// A message from the runtime to the consensus engine. This MUST NOT be
+		/// generated by the native code of any consensus engine, but this is
+		/// caught at compile time. The `ConsensusEngineId` is that of the
+		/// consensus engine for which this digest is intended. Consensus
+		/// engines MUST ignore digests with `ConsensusEngineId`s other than
+		/// their own.
+		Consensus((ConsensusEngineId, Vec<u8>), (a, b)),
+		/// Any 'non-system' digest item, opaque to the native code.  Runtimes
+		/// MUST verify these, and reject any they did not produce.  These MUST
+		/// NOT be produced by native code.
+		Other((Vec<u8>), (a)),
+	}
+
+	/// A digest item that is reserved for system use. Consensus engines and
+	/// runtimes **MUST NOT** push digests in this variant. This **SHOULD** be
```

Comment:
Currently both `AuthoritiesChange` && `ChangesTrieRoot` are generated && pushed by runtime.

Reply:
I clarified that these are reserved for the SRML.